components/style/rule_tree/mod.rs

```diff
@@ -533,9 +533,9 @@ impl StrongRuleNode {
     }
 
     fn next_sibling(&self) -> Option<WeakRuleNode> {
-        // FIXME(emilio): Investigate what ordering can we achieve without
-        // messing things up.
-        let ptr = self.get().next_sibling.load(Ordering::SeqCst);
+        // Note that relaxed is fine because we take care of racing if needed
+        // with the correct memory ordering at the end of our search.
```

Comment:
Sorry for the lag here, I've reworded this comment to be clear. What I meant is that we'll eventually arrive into the CAS-loop safely in the only case the ordering may actually impact anything as far as I know.